### ISSUE TITLE
Issue with apollo client.

### ISSUE THREAD

**Issue by halallahh** [join-monster/join-monster#26]
I tried query with apollo client for graphql server that use join-monster and I got this error:
> "GraphQL error: Cannot read property 'type' of undefined"

![error](https://s3.postimg.org/543zml95v/errorrrr.png)

Not working with my own setup and [join-monster-demo](https://github.com/stems/join-monster-demo).  I don't know this is problem with join monster or apollo client itself.

My apollo code look like this:
```javascript

class App extends React.Component {
    render() {
         const { users } = this.props.data;
         console.log(users)
         return (
                // view here
         );
   }
}

const USER = gql`
{
  users {
    email
    fullName
  }
}
`;

const withData = graphql(USER)(App);

```

**Comment by acarl005** [comment]
This is most likely a problem in Join Monster.

Can you post:
1. the actual GraphQL query that is getting received by the server?
2. your schema definition?

I want to try to reproduce this error. Also, if you have a public repo with the source code that I could check out, that would be great.

**Comment by halallahh** [comment]
Query works with GraphiQL but not for apollo client.

![graphiql](https://s13.postimg.org/ls362fzaf/Screen_Shot_2016_11_05_at_3_20_51_PM.png)

**Todo:**

``` javascript
const TodoType = new GraphQLObjectType({
    name: 'Todo',
    sqlTable: 'todos',
    uniqueKey: 'id',
    fields: () => ({
        id: {
            type: GraphQLInt,
        },
        task: {
            type: GraphQLString,
        },
        completed: {
            type: GraphQLBoolean,
        },
        comments: {
            type: new GraphQLList(CommentType),
            sqlJoin: (todoTable, commentTable) => `${todoTable}.id = ${commentTable}.todo_id`
        }
    })
});
```

**Comment**:

``` javacript
const CommentType = new GraphQLObjectType({
    name: 'Comment',
    sqlTable: 'comments',
    uniqueKey: 'id',
    fields: () => ({
        id: {
            type: GraphQLInt,
        },
        title: {
            type: GraphQLString,
        }
    })
});
```

**Root**:

``` javascript
const RootQuery = new GraphQLObjectType({
    name: 'Query',
    fields: () => ({
          todos: {
            type: new GraphQLList(TodoType),
            resolve: (parent, args, context, resolveInfo) => {
                return joinMonster(resolveInfo, context, sql => {
                    return knex.raw(sql);
                });
            }
        },
    })
});
```

**Comment by acarl005** [comment]
### Fixed in v0.5.1

The problem was that the apollo client adds some **introspection** to your query. This is what apollo was actually sending to the server:

```
{
  todos {
    task
    comments {
      title
      __typename
    }
    __typename
  }
}
```

The problem was that Join Monster tried to look for a user-defined `__typename` field. The internal fields for introspection should be ignored by Join Monster.

This is fixed. It should now work without modifying your code.

**Comment by halallahh** [comment]
oh, i see.
